src/shapes.rs

```diff
@@ -1,66 +1,57 @@
-use pyo3::{prelude::*, types::PyBytes};
-use std::borrow::Borrow;
+use pyo3::prelude::*;
 use std::fs::File;
-use std::io::{self, prelude::*};
+use std::io;
 use std::path::Path;
+use std::{borrow::Borrow, convert::TryInto};
 
-use geos::{CoordSeq, Geom, Geometry, PreparedGeometry};
+use geo::{point, Contains, Geometry, MultiPolygon, Point, Polygon, PreparedGeometry, Relate};
```

Comment:
This is a great test case! I'm curious how the perf cares for your use case with geo vs. geos.

(I'll take a look at your PR soon!)

Reply:
Thanks! I didnt get to work further on the pr yet.